src/schema/facet.rs

```diff
@@ -137,39 +181,7 @@ impl Borrow<str> for Facet {
 
 impl<'a, T: ?Sized + AsRef<str>> From<&'a T> for Facet {
     fn from(path_asref: &'a T) -> Facet {
-        #[derive(Copy, Clone)]
```

Comment:
We should probably remove the `From` implementation and possibly add a `TryFrom`.

Reply:
Sounds good. Is that necessary to do in this PR? or a separate PR is better?

Reply:
Can you do this in this PR?

Reply:
It just turned out that a generic TryFrom cannot be implemented in the way From is implemented here.  See the following thread for detail.  Actually I've succeeded for a specific implementation for `String` but I don't think it does ever have any use.

https://github.com/rust-lang/rust/issues/50133 (the reason is detailed in https://github.com/rust-lang/rust/issues/50133#issuecomment-488512355)

I would suggest keeping this as is now.

Reply:
We should consider replacing TryFrom with a custom trait. Would be ok from my side to merge it like that.

Reply:
@PSeitz @moriyoshi ...

Merging. We cannot decenlty ask you to fix all our mistakes :). Thank you!